x/mongo/driver/topology/server.go

```diff
@@ -707,28 +725,34 @@ func (s *Server) check() (description.Server, error) {
 	var err error
 	var durationNanos int64
 
+	previousDescription := s.Description()
+	streamable := previousDescription.TopologyVersion != nil
+
+	isNilConn := s.conn == nil
+	if !isNilConn {
+		s.publishServerHeartbeatStartedEvent(s.conn.ID(), s.conn.getCurrentlyStreaming() || streamable)
```

Comment:
Why are we publishing `ServerHeartbeatStarted` earlier in the `check` function? In theory this seems fine to me, but were you not seeing heartbeat started events for the initial heartbeat or something?